accepted/2020/feature-switch.md

```diff
@@ -0,0 +1,113 @@
+# Feature switch
+The functionality available in the .NET Core frameworks is getting bigger and wider with every release. Applications typically don't need or want all of it. With technologies like the linker we have the ability to remove parts of the framework which the app doesn't use. These technologies have limitations base don existing public API patterns and code behavior, sometimes they can't determine what the app really needs. This proposal adds a way to explicitly disable functionality in the framework.
+
+## Goals
+* Ability to declaratively turn-off functionality in the framework (and potentially any library/app)
+* Same behavior all the time - regardless of linker being used or not (`Debug` build behaves the same as published `Release` app).
+* Ability to set different defaults based on the type of application (Blazor, Xamarin, …)
+
+
+## Expected usage
+### Size
+By disabling functionality which is otherwise always available the linker can trim more code from the app reducing its size.
+For example
+* Disable support for FTP in [`HttpRequest.Create`](https://docs.microsoft.com/en-us/dotnet/api/system.net.webrequest.create?view=netcore-3.1#System_Net_WebRequest_Create_System_String_) (this is an example of a factory API which based on a string picks up features)
```

Comment:
A nit on this specific example: I'd rather see the app stop using the legacy WebRequest entirely ;)

Reply:
I know :-) - but it's such a nice example. I'd be happy to use a different one if we have it.

Reply:
Ok ;) There are a variety of factories or APIs that take options that configure what concrete instances are created, e.g. XmlReader/Writer.Create, Regex w/ RegexOptions.Compiled, etc.

Reply:
Good - I'll use Regex as an example.

Reply:
https://github.com/dotnet/runtime/blob/a83598dd151628672b1d2ff72310cc8c2065a019/src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs#L77-L81